metadata/src/utils/retain.rs

```diff
@@ -281,7 +358,6 @@ pub fn retain_metadata<F, G>(
 mod tests {
```

Comment:
Is there any test that we can add which ensures we don't strip outer enum variants when it would lead to breakage (ie they are used somewhere)?

Reply:
added a test that references initial issue